x-pack/plugins/enterprise_search/public/applications/app_search/components/search_ui/search_ui_logic.ts

```diff
@@ -84,7 +86,12 @@ export const SearchUILogic = kea<MakeLogicType<SearchUIValues, SearchUIActions>>
   listeners: ({ actions }) => ({
     loadFieldData: async () => {
       const { http } = HttpLogic.values;
-      const { engineName } = EngineLogic.values;
+      const { searchKey, engineName } = EngineLogic.values;
+
+      if (!searchKey) {
```

Comment:
We make a call to fetch all the data that the Search UI form needs below. As part of this API call, the API will actualy return an "errors" field that will return the message "It looks like you don't have any Public Search Keys with access to the 'engine1' engine. Please visit the Credentials page to set one up." In ent-search, we simply printed the error from that field to the screen.

In this UI, I opted to explicitly check for that condition and add that error message to an i18n string, so that it can be translated.

Reply:
I'm not sure if I agree with this approach.  It's seems strange to add this to listener that largely does an http call.  I think I would have kept this as a static part of the SearchUiForm with `useValues(EngineLogic)`

Reply:
True. I think I chose to put it in this logic file because I wanted to use our flash messaging system, which is stateful and requires a call to something like `setErrorMessage` to use.

I could have done that in like a useEffect in `SearchUiForm` but that seems clunkier. I like it here because we can short circuit the API call and all flash messages are generated from the logic file.

It could also be a stateless check where we just do like `!searchKey && <EuiCallout...`, but then that conflicts with the flash messaging system we use. Like it gets hairy having them both exist on the page together.